### ISSUE TITLE
Backend Error: The digest of missing repository fragment was found in a ticket member.

### ISSUE THREAD

**Issue by MaggieYingYi** [SNSystems/llvm-prepo#25]
Compile the following code:
```
namespace pstore {
template <typename ProcessPathFunction, typename BufferType>
int process_file_name(ProcessPathFunction, BufferType &p2) {
  p2.capacity();
}
template <typename, int = 256> class small_vector {
public:
  void capacity();
};
void process_file_name() {
  auto read_link = [] {};
  small_vector<char> buffer;
  process_file_name(read_link, buffer);
}
} // namespace pstore
```
using
```
$ rm clang.db
$ clang -O2 -c -target x86_64-pc-linux-gnu-repo  -std=gnu++11 -o test.o test.cpp
```
Compilation failed with the following error:
```
fatal error: error in backend: The digest of missing repository fragment 99effd5b5772b7cf34cc3b757a620c1c was found in a ticket member.
```

**Comment by MaggieYingYi** [comment]
This issue is happen when enabling the globalopt (global variable optimizer) pass. When global variable or function has been deleted, the corresponding metadata need to be removed from the repo.tickets metadata.

**Comment by MaggieYingYi** [comment]
This issue is also happen when enabling the globaldce (Dead Global Elimination) pass.

**Comment by MaggieYingYi** [comment]
Fixed in git commit: 76180d136fa1.
